theano/tensor/nlinalg.py

```diff
@@ -726,3 +726,37 @@ def norm(x, ord):
             raise ValueError(0)
     elif ndim > 2:
         raise NotImplementedError("We don't support norm witn ndim > 2")
+
+
+class LogAbsDet(Op):
+    """Computes the logarithm of absolute determinant of a square
+    matrix M, log(abs(det(M))), on CPU. Avoids det(M) overflow/
+    underflow.
+
+    TODO: add GPU code!
+    """
+    def make_node(self, x):
+        x = theano.tensor.as_tensor_variable(x)
+        o = theano.tensor.scalar(dtype=x.dtype)
+        return Apply(self, [x], [o])
+
+    def perform(self, node, inputs, outputs):
+        try:
+            (x,) = inputs
+            (z,) = outputs
+            s = numpy.linalg.svd(x, compute_uv=False)
+            log_abs_det = numpy.sum(numpy.log(numpy.abs(s)))
```

Comment:
We didn't document much making new gpu op as we are in transition to a new gpu back-end...

For a first pass, you can make one GPU op that have its gradient on the CPU and use it directly. Then after, you can add an optimization that will convert the CPU op to the GPU op (I can help, it is very easy when you know how to do it).

To make a GPU op, I would use as an example the code of CuFFTOp in sandbox/cuda/fftconv.py. This use scikits.cuda for computation, so you can reuse it for the svd implementation.